docs/activities.md

```diff
@@ -1,81 +1,69 @@
 
-One common use case for such leader election is monitoring. An activity executes an internal loop that periodically polls some API and checks for some condition. It also heartbeats on every iteration. If the condition is satisfied, the activity completes which lets its workflow to handle it. If the activity worker dies, the activity times out after the heartbeat interval is exceeded and is retried on a different worker. The same pattern works for polling for new files in Amazon S3 buckets or responses in REST or other synchronous APIs.
+### Heartbeat timeout
 
-## Cancellation
+Within Activity code, [Heartbeats](heartbeats) can be recorded. To record a Heartbeat is to ping the Temporal server letting it know that the Activity is still executing. This timeout sets a maximum duration between Heartbeats, effectively enabling manual timeout implementations for Activity executions. This timeout is optional and should only be used if Heartbeats are recorded within the Activity. For practical implementation guidance, read [How to timeout Activities and Workflows](timeout-activities-and-workflows).
 
-A workflow can request an activity cancellation. Currently the only way for an activity to learn that it was cancelled is through heart beating. The heartbeat request fails with a special error indicating that the activity was cancelled. Then it is up to the activity implementation to perform all the necessary cleanup and report that it is done with it. It is up to the workflow implementation to decide if it wants to wait for the activity cancellation confirmation or just proceed without waiting.
+### Wait for cancellation
 
-Another common case for activity heartbeat failure is that the workflow that invoked it is in a completed state. In this case an activity is expected to perform cleanup as well.
+When an Activity is cancelled it may be necessary for the Activity to revert actions that it has already taken. This option specifies to the Workflow whether to wait for the clean up to occur after an Activity has been cancelled.
 
-## Activity Task Routing through Task Queues
+### Activity Id
 
-Activities are dispatched to workers through task queues. Task queues are queues that workers listen on. Task queues are highly dynamic and lightweight. They don't need to be explicitly registered. And it is okay to have one task queue per worker process. It is normal to have more than one activity type to be invoked through a single task queue. And it is normal in some cases (like host routing) to invoke the same activity type on multiple task queues.
+Use this to give your Activity a name or unique identifier that correlates to your business logic. This is optional and is not used by the Temporal server.
```

Comment:
1. "this"
2. I would use the word mapping and "relevant business ID" or something